glib-macros/src/lib.rs

```diff
@@ -265,6 +266,101 @@ pub fn clone(item: TokenStream) -> TokenStream {
     clone::clone_inner(item)
 }
 
+/// The same as [`gclosure_local!`](crate::gclosure_local) but uses [`Closure::new`] as a
+/// constructor.
+///
+/// [`Closure::new`]: https://gtk-rs.org/gtk-rs-core/stable/latest/docs/glib/closure/struct.Closure.html#method.new
+#[proc_macro]
+#[proc_macro_error]
+pub fn gclosure(item: TokenStream) -> TokenStream {
+    gclosure::gclosure_inner(item, "new")
+}
+
+/// Macro for creating a [`Closure`] object. This is a wrapper around
+/// [`Closure::new_local`] that type checks its arguments at runtime.
+///
+/// A `Closure` takes [`Value`] objects as inputs and output. This macro will
+/// automatically convert the inputs to Rust types when invoking its callback, and then will
+/// convert the output back to a `Value`. All inputs must implement the
+/// [`FromValue`] trait, and outputs must either implement the
+/// [`ToValue`] trait or be the unit type `()`.
```

Comment:
Might be worth mentioning here that this will panic if the closure is called with different types than what is used inside the macro